src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -711,6 +720,31 @@ private ConnectionState ChangeState(ConnectionState to)
             }
         }
 
+        private static bool IsWebSocketsSupported()
+        {
+            if (_hasWebSocketsSupport == null)
+            {
+#if NETCOREAPP2_1
+                // .NET Core 2.1 and above has a managed implementation
+                _hasWebSocketsSupport = true;
+#else
+                bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+                if (!isWindows)
+                {
+                    // Assume other OSes have sockets implementation
+                    _hasWebSocketsSupport = true;
+                }
+                else
+                {
+                    // Window 8 and above has web sockets
+                    _hasWebSocketsSupport = Environment.OSVersion.Version > new Version(6, 1);
+                }
+#endif
+            }
+
+            return _hasWebSocketsSupport.Value;
```

Comment:
I think we can skip the static. In the first place, `Nullable<bool>` is a two-field struct so it could shear, plus the code required to calculate it is basically just checking some already-cached values ([`Environment.OSVersion`](https://source.dot.net/#System.Runtime.Extensions/System/Environment.cs,137), [`RuntimeInformation.IsOSPlatform`](https://source.dot.net/#System.Runtime.InteropServices.RuntimeInformation/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Windows.cs,17))